src/components/cc-logs-controller/cc-logs-controller.js

```diff
@@ -0,0 +1,504 @@
+   */
+  _getDateDisplayLabel (dateDisplay) {
+    if (dateDisplay === 'none') {
+      return i18n('cc-logs-controller.date-display.none');
+    }
+    if (dateDisplay === 'datetime-iso') {
+      return i18n('cc-logs-controller.date-display.datetime-iso');
+    }
+    if (dateDisplay === 'time-iso') {
+      return i18n('cc-logs-controller.date-display.time-iso');
+    }
+    if (dateDisplay === 'datetime-short') {
+      return i18n('cc-logs-controller.date-display.datetime-short');
+    }
+    if (dateDisplay === 'time-short') {
+      return i18n('cc-logs-controller.date-display.time-short');
+    }
+  }
+
+  /**
+   *
+   * @param {Timezone} timezone
+   * @return {string}
+   */
+  _getTimezoneLabel (timezone) {
+    if (timezone === 'UTC') {
+      return i18n('cc-logs-controller.timezone.utc');
+    }
+    if (timezone === 'local') {
+      return i18n('cc-logs-controller.timezone.local');
+    }
+  }
+
+  /**
+   *
+   * @param {string} palette
+   * @return {string}
+   */
+  _getPaletteLabel (palette) {
+    if (palette === 'default') {
+      return i18n('cc-logs-controller.palette.default');
+    }
+    return palette;
+  }
+
+  /* endregion */
+
+  willUpdate (_changedProperties) {
+    if (_changedProperties.has('metadataRenderers') || _changedProperties.has('metadataDisplay')) {
+      this._resolvedMetadataRenderers = this._resolveMetadataRenderers(this.metadataRenderers, this.metadataDisplay);
+    }
+  }
+
+  render () {
+    return html`
+      <div class="header-slot">
+        <slot name="header"></slot>
+      </div>
+
+      <cc-button
+        class="header-scroll-button"
+        .icon=${followIcon}
+        accessible-name="${i18n('cc-logs-controller.scroll-to-bottom')}"
+        hide-text
+        @cc-button:click=${this._onScrollToBottomButtonClick}
+      ></cc-button>
+
+      <cc-popover
+        class="header-options-popover"
+        .icon=${optionsIcon}
+        accessible-name="${i18n('cc-logs-controller.show-logs-options')}"
+        hide-text
+        position="bottom-right"
+      >
+        ${this._renderOptions()}
+      </cc-popover>
+
+      <cc-logs-beta
+        class="logs"
+        ${ref(this._logsRef)}
+        .dateDisplay=${this.dateDisplay}
+        .filter=${this.filter}
+        ?follow=${this.follow}
+        .limit=${this.limit}
+        .logs=${this.logs}
+        .metadataRenderers=${this._resolvedMetadataRenderers}
+        ?strip-ansi=${this.stripAnsi}
+        style=${PALETTES[this.palette]}
+        .timezone=${this.timezone}
+        ?wrap-lines=${this.wrapLines}
+      ></cc-logs-beta>
+    `;
+  }
+
+  _renderOptions () {
+    return html`
+      <div class="options">
+        ${this._renderDisplayOptions()}
+        ${this._renderDateOptions()}
+        ${this._renderMetadataOptions()}
+      </div>
+    `;
+  }
+
+  _renderDisplayOptions () {
+    const PALETTE_CHOICES = Object.entries(PALETTES)
+      .map(([p]) => ({ label: this._getPaletteLabel(p), value: p }));
```

Comment:
question: can/should we move this to the beginning of the file (and do it just once)?

This also means removing the `_getPaletteLabel`.

We do something similar here: https://github.com/CleverCloud/clever-components/blob/master/src/components/cc-pricing-estimation/cc-pricing-estimation.js#L22-L33

Reply:
I don't want to call the i18n function outside the instance of the component.